microsoft/azure-databricks-operator
# Draft: Contribute load testing and mock Databricks API

# DRAFT PR

This is a PR to contribute back the load testing framework built by our dev crew (@stuartleeks @lawrencegripper @storey247 @martinpeck @damoodamoo @bart-jansen) at Microsoft.

This PR contains:
- Locust load testing framework + load test scenarios for the Databricks Operator
- Databrick Mock API used to run the load tests against
- Install python and initalize virutal enviroment in dev container
- Load testing and MockAPI dashboards in grafana
- Make commands to:
   - test and run Locust, mockAP
   - run load tests in a kind cluster
- Docs for MockAPI and Locust usage
- Go script to verify load tests have run successfully
- Run the load tests in the build pipeline
